v2/tools/generator/internal/codegen/pipeline/implement_importable_resource_interface.go

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/dave/dst"
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astbuilder"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/config"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/conversions"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/functions"
+)
+
+const ImplementImportableResourceInterfaceStageID = "implementImportableResourceInterface"
+
+func ImplementImportableResourceInterface(
+	configuration *config.Configuration,
+	idFactory astmodel.IdentifierFactory) *Stage {
+	stage := NewStage(
+		ImplementImportableResourceInterfaceStageID,
+		"Implement the ImportableResource interface for resources that support import via asoctl",
+		func(ctx context.Context, state *State) (*State, error) {
+			defs := state.Definitions()
+
+			// Scan for the resources the ImportableResource interface injected
+			scanner := newSpecInitializationScanner(state.Definitions(), configuration)
+			rsrcs, err := scanner.findResources()
+			if err != nil {
+				return nil, errors.Wrapf(err, "unable to find resources that support import")
+			}
+
+			injector := astmodel.NewInterfaceInjector()
+
+			errs := make([]error, 0, 10)
```

Comment:
Maybe make this a dynamic array?

Reply:
😕 

This _**is**_ a dynamic slice - size zero, capacity 10. 
Our performance linter triggers if you don't specify a capacity up front, as a part of work that @Porges did ages ago.

Reply:
Ok, I'll change these.